tests/baselines/reference/baseClassImprovedMismatchErrors.errors.txt

```diff
@@ -17,7 +17,7 @@ baseClassImprovedMismatchErrors.ts(14,5): error TS2416: Property 'n' in type 'De
           Type 'string | number' is not assignable to type 'number'.
             Type 'string' is not assignable to type 'number'.
 baseClassImprovedMismatchErrors.ts(15,5): error TS2416: Property 'fn' in type 'DerivedInterface' is not assignable to the same property in base type 'Base'.
-  Type '() => string | number' is not assignable to type '() => number'.
+  Type '() => number | string' is not assignable to type '() => number'.
     Type 'string | number' is not assignable to type 'number'.

```

Comment:
Yeah, the signature has a declaration, so can reuse its' return type node, but the union type itself has no such luck. Now, it doesn't *have* to be that way (intersections remember their order forever because it matters if there are overloads, unions could do the same even if it only matters for looks) - but we'd produce more type identities.